### PR TITLE
Minor changes in is_matrix, in_list, removal of duplicate definitions and format

### PR DESCRIPTION
**Pull request by RonaldoCMP**

### CODE REVIEW

**Comment by RonaldoCMP** [comment]
Corrections

File: math.scad [hunk 1]
```diff
@@ -778,21 +778,18 @@ function back_substitute(R, b, transpose = false) =
...
-      R[ind][ind] == 0 ? []
-    : let(
-          newvalue = len(x)==0
-            ? b[ind]/R[ind][ind]
-            : (b[ind]-select(R[ind],ind+1,-1) * x)/R[ind][ind]
-      )
-      _back_substitute(R, b, concat([newvalue],x));
+    ?   let( R = [for(i=[0:n-1]) [for(j=[0:n-1]) R[n-1-j][n-1-i]]] )
+        reverse( back_substitute( R, reverse(b), x ) )  
+    :   len(x) == n ? x :
```
Comment (revarbat) [inline math.scad:783]: Regressions fail on this line.  The variable `x` is not yet declared.

File: math.scad [hunk 2]
```diff
@@ -240,7 +240,7 @@ function atanh(x) =
 //   quant([9,10,10.4,10.5,11,12],3);      // Returns: [9,9,9,12,12,12]
 //   quant([[9,10,10.4],[10.5,11,12]],3);  // Returns: [[9,9,9],[12,12,12]]
 function quant(x,y) =
-    assert(is_finite(y) && !approx(y,0,eps=1e-24), "The multiple must be a non zero number.")
+    assert(is_finite(y) && !approx(y,0,eps=1e-24), "The multiple must be a non zero integer.")
```
Comment (revarbat) [inline math.scad:243]: Is NOT required to be an integer.

File: math.scad [hunk 3]
```diff
@@ -272,7 +272,7 @@ function quant(x,y) =
 //   quantdn([9,10,10.4,10.5,11,12],3);      // Returns: [9,9,9,9,9,12]
 //   quantdn([[9,10,10.4],[10.5,11,12]],3);  // Returns: [[9,9,9],[9,9,12]]
 function quantdn(x,y) =
-    assert(is_finite(y) && !approx(y,0,eps=1e-24), "The multiple must be a non zero number.")
+    assert(is_finite(y) && !approx(y,0,eps=1e-24), "The multiple must be a non zero integer.")
```
Comment (revarbat) [inline math.scad:275]: Is NOT required to be an integer.

File: math.scad [hunk 4]
```diff
@@ -304,7 +304,7 @@ function quantdn(x,y) =
 //   quantup([9,10,10.4,10.5,11,12],3);      // Returns: [9,12,12,12,12,12]
 //   quantup([[9,10,10.4],[10.5,11,12]],3);  // Returns: [[9,12,12],[12,12,12]]
 function quantup(x,y) =
-    assert(is_finite(y) && !approx(y,0,eps=1e-24), "The multiple must be a non zero number.")
+    assert(is_finite(y) && !approx(y,0,eps=1e-24), "The multiple must be a non zero integer.")
```
Comment (revarbat) [inline math.scad:307]: Is NOT required to be an integer.

File: math.scad [hunk 5]
```diff
@@ -1047,7 +1037,7 @@ function count_true(l, nmax) =
 //   data[len(data)-1].  This function uses a symetric derivative approximation
 //   for internal points, f'(t) = (f(t+h)-f(t-h))/2h.  For the endpoints (when closed=false) the algorithm
 //   uses a two point method if sufficient points are available: f'(t) = (3*(f(t+h)-f(t)) - (f(t+2*h)-f(t+h)))/2h.
-//   .
+//
```
Comment (revarbat) [inline math.scad:1040]: For docs generation, the single . syntax is correct and easier to see.

**Pull request merged**